python/chemiscope/explore.py

```diff
@@ -0,0 +1,145 @@
+import os
+
+from dscribe.descriptors import SOAP
+from sklearn.decomposition import PCA
+
+from .jupyter import show
+
+
+def explore(frames, featurize=None, properties=None, mode="default"):
+    """
+    Visualize the dataset with dimensionality reduction.
+    If no function is provided as a `featurize` argument, a default SOAP and PCA based
+    featurizer is used. SOAP parameters (e.g., cutoff radius, number of radial and
+    angular functions, etc.) are predefined. The SOAP computation uses all available
+    CPU cores for parallelization. PCA reduces the dimensionality to two components
+    by default.
+
+    :param list frames: list of ASE Atoms objects
+
+    :param callable featurize: optional. Function to compute features and perform
+        dimensionality reduction on the `frames`. The function should take `frames` as
+        input and return a 2D array of reduced features. If `None`, a default SOAP and
+        PCA based featurizer is used.
+
+    :param dict properties: optional. Additional properties to be included in the
+        visualization. This dictionary can contain any other relevant data associated
+        with the atomic structures. Properties can be extracted from frames with
+        :py:func:`extract_properties` or manually defined by the user.
+
+    :param str mode: optional. Visualization mode for the chemiscope widget. Can be one
+        of "default", "structure", or "map". The default mode is "default".
+
+    :return: a chemiscope widget for interactive visualization
+
+    Here is an example of usage with and without providing a function to do the
+    reprensentation and reduction. The frames are obtained by reading the structures
+    from a file that `ase <ase-io_>`_ can read, and performing PCA using `sklearn`_ on
+    a descriptor computed with SOAP with `dscribe`_ library.
+
+    .. code-block:: python
+
+        import chemiscope
+        from ase.io import read
+        from dscribe.descriptors import SOAP
+        from sklearn.decomposition import KernelPCA
```

Comment:
```suggestion
        import ase.io
        import dscribe.descriptors
        import sklearn.decomposition
```

and then using explicit path makes it clearer where things come from